docs/gatsby-node.js

```diff
@@ -33,14 +41,33 @@ exports.createPages = async ({ graphql, actions, reporter }) => {
     reporter.panicOnBuild(`Error when querying guides.`);
     return;
   }
-  result.data.allMdx.nodes.forEach(guide => {
+
+  result.data.allMdx.nodes.forEach(({ id, frontmatter, slug, body }) => {
     createPage({
-      path: `/guides/${path.basename(guide.slug)}`,
+      path: `/guides/${path.basename(slug)}`,
       component: `${__dirname}/src/templates/Guide.tsx`,
       context: {
-        title: guide.frontmatter.title,
-        body: guide.body,
+        id,
+        slug,
+        body,
+        ...frontmatter,
       },
     });
   });
 };
+
+exports.createSchemaCustomization = ({ actions }) => {
+  actions.createTypes(
+    `
+    type Mdx implements Node @infer {
```

Comment:
I think if you want to infer types, you don't need to say so explicitly.

Also, I think it might be better to not infer types. At the very least, I would think it'd be good not infer them for the frontmatter and only include things where we have explicitly typed the fields.